.github/workflows/static-checks-npm.yml

```diff
@@ -65,3 +77,26 @@ jobs:
           base: ""
           head: ${{ github.ref_name }}
           extra_args: ${{ inputs.trufflehog_extra_args }}
+  scan-vulns:
+    if: |
+      (inputs.enable_semgrep_action == true) &&
+      (github.actor != 'dependabot[bot]')
+    name: Vulnerability scanning
+    permissions:
+      security-events: write
+    runs-on: ubuntu-latest
+    container:
+      image: semgrep/semgrep:1.142.1-nonroot
+    steps:
+      - uses: actions/checkout@v5
+        with:
+          fetch-depth: 0
+      - name: Semgrep CE scan
+        run: |
+          semgrep ${{ inputs.semgrep_extra_args }} --sarif --sarif-output=${{ inputs.semgrep_sarif_path }}
+      - name: Upload Semgrep's SARIF results
+        uses: github/codeql-action/upload-sarif@v3
+        if: always()
+        with:
+          sarif_file: ${{ inputs.semgrep_sarif_path }}
+          category: semgrep
```

Comment:
We probably want to offer the following behaviour

semgrep_upload_type either sarif, artefact or none.  And then change or skip the upload step accordingly.  We should do this on its dedicated workflow as well.